src/Compatibility/Microsoft.DotNet.ApiSymbolExtensions/AssemblySymbolLoader.cs

```diff
@@ -51,6 +51,48 @@ public class AssemblySymbolLoader : IAssemblySymbolLoader
         /// </summary>
         public const string AssemblyReferenceNotFoundErrorCode = "CP1002";
 
+        /// <summary>
+        /// Creates an assembly symbol loader and its corresponding assembly symbols from the given DLL files in the filesystem.
+        /// </summary>
+        /// <param name="log">The logger instance to use for message logging.</param>
+        /// <param name="assembliesPaths">A collection of paths where the assembly DLLs should be searched.</param>
+        /// <param name="assemblyReferencesPaths">An optional collection of paths where the assembly references should be searched.</param>
+        /// <param name="respectInternals">Whether to include internal symbols or not.</param>
+        /// <returns>A tuple containing an assembly symbol loader and its corresponding dictionary of assembly symbols.</returns>
+        public static (AssemblySymbolLoader, Dictionary<string, IAssemblySymbol>) CreateFromFiles(ILog log, string[] assembliesPaths, string[]? assemblyReferencesPaths, bool respectInternals = false)
```

Comment:
I'm not sure of the actual code flow, but do you need to make sure ILog has a value here? Doesn't look like the constructor of `AssemblySymbolLoader` checks it either.

Reply:
Since it's a non-nullable argument, the callers should always pass a valid instance.